Zig.gitignore

```diff
@@ -0,0 +1,6 @@
+# Object files
+*.o
+
```

Comment:
Based on the thread on #3442, I think they only want the zig-cache and zig-out lines. 

`zig build` should put all the .o files in zig-out, yes?

Reply:
Yes - fixed